components/form/HookOption.vue

```diff
@@ -0,0 +1,271 @@
+
+    const defaultExec = { exec: { command: [] } };
+    const defaultHttpGet = {
+      httpGet: {
+        host:        '',
+        path:        '',
+        port:        null,
+        scheme:      '',
+        httpHeaders: [{ name: '', value: '' }]
+      }
+    };
+    const defaultTcpSocket = { tcpSocket: { host: '', port: null } };
+
+    return {
+      lifecycle: this.value,
+      httpGetOptions,
+      tcpSocketOptions,
+      selectHook,
+      defaultExec,
+      defaultHttpGet,
+      defaultTcpSocket
+    };
+  },
+
+  computed: {
+    isView() {
+      return this.mode === _VIEW;
+    },
+  },
+
+  created() {
+    this.queueUpdate = debounce(this.update, 500);
+  },
+
+  methods: {
+    addHeader() {
+      const header = { name: '', value: '' };
+
+      this.lifecycle.httpGet.httpHeaders.push(header);
+    },
+
+    removeHeader(index) {
+      this.lifecycle.httpGet.httpHeaders.splice(index, 1);
+      this.queueUpdate();
+    },
+
+    update() {
+      const { ...leftovers } = this.lifecycle;
+
+      switch (this.selectHook) {
+      case 'exec':
+        this.deleteLeftovers(leftovers);
+        Object.assign(this.lifecycle, this.defaultExec);
+        break;
+      case 'httpGet':
+        this.deleteLeftovers(leftovers);
+        Object.assign(this.lifecycle, this.defaultHttpGet);
+        break;
+      case 'tcpSocket':
+        this.deleteLeftovers(leftovers);
+        Object.assign(this.lifecycle, this.defaultTcpSocket);
+        break;
+      default:
+        break;
+      }
+    },
+
+    deleteLeftovers(leftovers) {
+      if (leftovers) {
+        for (const obj in leftovers) {
+          delete this.lifecycle[obj];
+        }
+      }
+    }
+  }
+};
+</script>
+
+<template>
+  <div @input="update">
+    <div class="row mb-10">
+      <RadioGroup
+        v-model="selectHook"
+        name="selectHook"
+        :options="['exec', 'httpGet', 'tcpSocket']"
+        :labels="[
+          t('workload.container.lifecycleHook.exec.add'),
+          t('workload.container.lifecycleHook.httpGet.add'),
+          t('workload.container.lifecycleHook.tcpSocket.add')
+        ]"
+        :mode="mode"
+        @input="update"
+      />
+    </div>
+
+    <template v-if="selectHook === 'exec'">
+      <div class="mb-20 single-value">
+        <h4>{{ t('workload.container.lifecycleHook.exec.title') }}</h4>
+        <div>
+          <ShellInput
+            v-model="lifecycle.exec.command"
+            :mode="mode"
+            :label="t('workload.container.lifecycleHook.exec.command.label')"
+            :placeholder="t('workload.container.lifecycleHook.exec.command.placeholder')"
+          />
+        </div>
+      </div>
+    </template>
+
+    <template v-if="selectHook === 'httpGet'">
+      <template>
+        <h4>{{ t('workload.container.lifecycleHook.httpGet.title') }}</h4>
+        <slot name="httpGetOption">
+          <div class="var-row">
+            <div v-for="option in httpGetOptions" :key="option.vKey">
```

Comment:
I think the scheme httpGetOption should use a dropdown since there appears to only be two valid choices: when I tried to define a HTTP request lifecycle hook I got the following error: 
`Deployment.apps "test-lifecycle" is invalid: spec.template.spec.containers[0].lifecycle.preStop.httpGet.scheme: Unsupported value: "https": supported values: "HTTP", "HTTPS"`

You could continue the v-for pattern you have going on with a LabeledSelect that has ``v-else-if="option.value==='scheme'"` ` but I think if you're going to end up putting 3 inputs into the template anyway it would be better to get rid of the httpGetOptions array and just template out all four input fields.

Reply:
I changed `httpGet` to be a template with all 4 input fields and removed `httpGetOptions`. The `scheme` is now a dropdown with the two supported values.